### PR TITLE
Clone git submodules

### PR DESCRIPTION
**Pull request by luv**
With this simple change, git submodules are cloned so they can be used right away.

### CODE REVIEW

**Comment by luv** [comment]
Sounds resonable. I can imagine that changing the current behaviour of git clone might surprise a few users.

What new command name would you recommend for clone with --recurse ?

**Comment by randy3k** [comment]
Maybe `git clone recursively`?

**Comment by luv** [comment]
I have updated the PR to use command "git: clone recursively".

**Comment by randy3k** [comment]
Lgtm

File: core/commands/init.py [hunk 1]
```diff
@@ -118,7 +119,10 @@ def on_enter_directory(self, path):
 
     def do_clone(self):
         self.window.status_message("Start cloning {}".format(self.git_url))
-        self.git("clone", self.git_url, self.suggested_git_root, working_dir='.')
```
Comment (stoivo) [inline core/commands/init.py]: You can do to dry it up and nicer when we want to add more clone options
```python
self.git(
    "clone", 
    "--recursive" if self.recursive else None
    self.git_url, 
    self.suggested_git_root, 
    working_dir='.')
```

Comment (luv) [reply core/commands/init.py]: sure

Comment (stoivo) [reply core/commands/init.py]: Use `else None`, Everything else is good

Comment (luv) [reply core/commands/init.py]: sure

**Comment by stoivo** [comment]
Thanks

**Pull request merged**